apps/arweave/src/ar_nonce_limiter_client.erl

```diff
@@ -92,6 +97,23 @@ handle_cast(pull, State) ->
 			end
 	end;
 
+handle_cast(request_sessions, State) ->
+	#state{ remote_servers = Q } = State,
+	{{value, RawPeer}, Q2} = queue:out(Q),
+	State2 = State#state{ remote_servers = queue:in(RawPeer, Q2) },
+	case resolve_peer(RawPeer) of
+		error ->
+			%% Push the peer to the back of the queue.
+			{noreply, State2};
+		{ok, Peer} ->
+			case fetch_and_apply_session_and_previous_session(Peer) of
+				ok ->
+					{noreply, State};
```

Comment:
Is there any risk that the VDF server will respond with a stale session because it itself has gotten stuck? Or perhaps it's a slow VDF server? i.e. is there any concern that we might get stuck always asking the same slow VDF server for the session and never rotate the queue?

Reply:
If the server is stuck, the function should return a non-ok response, e.g., a #nonce_limiter_update_response with a step number indicating we are ahead. Good catch, I've found and patched two places where [it was not the case](https://github.com/ArweaveTeam/arweave/pull/570/commits/538b5978787f0acebe1fd4ce6a1f3a90dc85d16f).

Reply:
makes sense!